src/main/java/org/mitre/hapifhir/MedMorphToCIBMTR.java

```diff
@@ -155,10 +194,29 @@ protected void postBundle(String authToken, String ccn, List<BundleEntryComponen
     }
   }
 
-  protected JSONArray getObservationEntries(String ccn, List<BundleEntryComponent> observationEntries, String resourceId) {
+  protected JSONArray getObservationEntries(CloseableHttpClient httpClient, String authToken, String ccn, List<BundleEntryComponent> observationEntries, String resourceId, boolean isPatientNew) throws Exception {
     JSONArray entryArray = new JSONArray();
 
     for (BundleEntryComponent entry : observationEntries) {
+      // If observation already exists on server, skip posting of resource
+      if (!entry.hasFullUrl()) continue;
+      String fullUrl = entry.getFullUrl();
+
+      // Only check if patient isn't new
+      if (!isPatientNew) {
+        HttpGet httpGet = new HttpGet(cibmtrUrl + "Observation?identifier=" + fullUrl);
```

Comment:
It won't let me add a comment to the previous thread, but upon closer inspection their API might not support this. On page 11 of the doc it lists querying observations by CRID but nothing else.
![image](https://user-images.githubusercontent.com/13512036/135461542-aacda0b4-c513-4e8f-8806-1ca45e3b049f.png)
Still I would say hold off until we get access to a test environment to confirm their capabilities. Switching to fetching all resources involves other challenges like pagination